packages/material-ui/src/BottomNavigationAction/BottomNavigationAction.test.js

```diff
@@ -1,108 +1,81 @@
-    const wrapper = mount(<BottomNavigationAction icon={icon} showLabel={false} />);
-    const root = findOutermostIntrinsic(wrapper);
-    assert.strictEqual(root.hasClass(classes.root), true);
-    assert.strictEqual(root.hasClass(classes.iconOnly), true);
+  it('removes the `iconOnly` class when `selected`', () => {
+    const { getByRole } = render(<BottomNavigationAction selected />);
+
+    expect(getByRole('button')).not.to.have.class(classes.iconOnly);
   });
 
-  it('should render icon', () => {
-    const wrapper = mount(<BottomNavigationAction icon={icon} />);
-    assert.strictEqual(wrapper.contains(icon), true);
+  it('removes the `iconOnly` class when `showLabel`', () => {
+    const { getByRole } = render(<BottomNavigationAction showLabel />);
+
+    expect(getByRole('button')).not.to.have.class(classes.iconOnly);
   });
 
-  it('should render label with the selected class', () => {
-    const wrapper = mount(<BottomNavigationAction icon={icon} selected />);
-    const labelWrapper = wrapper.find(`.${classes.label}`).first();
-    assert.strictEqual(labelWrapper.exists(), true);
-    assert.strictEqual(labelWrapper.hasClass(classes.selected), true);
+  it('should render the passed `icon`', () => {
+    const { getByRole } = render(<BottomNavigationAction icon={<div data-testid="icon" />} />);
+
+    expect(within(getByRole('button')).getByTestId('icon')).to.be.ok;
   });
 
-  it('should render label with the iconOnly class', () => {
-    const wrapper = mount(<BottomNavigationAction icon={icon} showLabel={false} />);
-    const labelWrapper = wrapper.find(`.${classes.label}`).first();
-    assert.strictEqual(labelWrapper.exists(), true);
-    assert.strictEqual(labelWrapper.hasClass(classes.iconOnly), true);
+  it('should render label with the selected class when selected', () => {
+    const { container } = render(<BottomNavigationAction selected />);
+
+    expect(container.querySelector(`.${classes.label}`)).to.have.class(classes.selected);
   });
 
-  it('should not render an Icon if icon is not provided', () => {
-    const wrapper = mount(<BottomNavigationAction />);
-    assert.strictEqual(wrapper.find(Icon).exists(), false);
+  it('should render label with the `iconOnly` class', () => {
+    const { container } = render(<BottomNavigationAction icon />);
+
+    expect(container.querySelector(`.${classes.label}`)).to.have.class(classes.iconOnly);
   });
 
   describe('prop: onClick', () => {
     it('should be called when a click is triggered', () => {
       const handleClick = spy();
-      const wrapper = mount(
-        <BottomNavigationAction icon="book" onClick={handleClick} value="foo" />,
-      );
-      wrapper.simulate('click', 'bar');
-      assert.strictEqual(handleClick.callCount, 1);
-    });
-  });
+      const { getByRole } = render(<BottomNavigationAction onClick={handleClick} />);
 
-  describe('prop: onChange', () => {
-    it('should be called when a click is triggered', () => {
```

Comment:
Classic example for misleading tests. Clicking a button doesn't trigger change on that button.